spec/preservation_ingest/update_catalog_spec.rb

```diff
@@ -0,0 +1,54 @@
+describe Robots::SdrRepo::PreservationIngest::UpdateCatalog do
+  subject(:update_catalog_obj) { described_class.new }
+
+  let(:bare_druid) { 'bj102hs9687' }
+  let(:size) { 2342 }
+  let(:version) { 3 }
+  let(:strg_root) { "some/storage/location/from/endpoint/table" }
+  let(:faraday_dbl) { class_double(Faraday) }
+  let(:url) { 'http://localhost:3000' }
+  let(:args) do
+    { druid: bare_druid, incoming_version: version, incoming_size: size, storage_location: strg_root }
+  end
+  let(:mock_so) do
+    instance_double(Moab::StorageObject, object_pathname: instance_double(Pathname), size: size, storage_root: strg_root, current_version_id: version)
+  end
+
+  before do
+    allow(Moab::StorageServices).to receive(:find_storage_object).and_return(mock_so)
+  end
+
+  describe '#perform' do
+    before do
+      allow(LyberCore::Log).to receive(:debug).with(any_args)
+      allow(Faraday).to receive(:new).with(url: url).and_return(faraday_dbl)
+    end
+
+    context 'object is new' do
+      it 'POST to /catalog' do
+        response = instance_double(Faraday::Response, status: 201)
+        expect(faraday_dbl).to receive(:post).with('/catalog', args).and_return(response)
+        expect(faraday_dbl).not_to receive(:patch)
+        update_catalog_obj.perform(bare_druid)
+      end
```

Comment:
I think you'll be re-writing these tests with my comments above, but note that you should also have a test for the error conditions.

- when object is new, and pres cat gives  xxx code
- when object gets new version and pres cat gives  xxx code

- when faraday gives an error and retries deals w it
- when faraday gives an error beyond retries.